### PR TITLE
[AKS] `az aks create/update`: Add support for KEDA workload auto-scaler

### PR DESCRIPTION
**Pull request by v-shenoy**
**Related command**

```
az aks create
az aks update
```

**Description**

Allows AKS customers to enable / disable the KEDA addon for workload autoscaling.

**Testing Guide**

**History Notes**

[AKS] `az aks create`: Add new parameter `--enable-keda` to enable KEDA workload auto-scaler
[AKS] `az aks update`: Add new parameters `--enable-keda`/`--disable-keda` to enable/disable KEDA workload auto-scaler

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by FumingZhang** [comment]
Sync changes from PR [#4905](https://github.com/Azure/azure-cli-extensions/pull/4905) in aks-preview.

File: src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py [hunk 1]
```diff
@@ -5160,6 +5232,39 @@ def set_up_oidc_issuer_profile(self, mc: ManagedCluster) -> ManagedCluster:
...
+        self._ensure_mc(mc)
+
+        if self.context.get_enable_keda():
+            if mc.workload_auto_scaler_profile is None:
+                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoscalerProfile()
+            mc.workload_auto_scaler_profile.keda = ManagedClusterWorkloadAutoscalerProfileKeda(enabled=True)
+
+        return mc
+
+    def update_workload_auto_scaler_profile(self, mc: ManagedCluster) -> ManagedCluster:
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py:5249]: You should move this function to the other class `AKSManagedClusterUpdateDecorator`. The function to construct the updated profile (`update_mc_profile_default`) is also in this class.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -7938,3 +7938,70 @@ def test_aks_nodepool_add_with_gpu_instance_profile(self, resource_group, resour
...
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'location': resource_group_location,
+            'ssh_key_value': self.generate_ssh_keys(),
+        })
+
+        # create: enable-keda
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --location={location} --ssh-key-value={ssh_key_value} --output=json ' \
+                     '--aks-custom-headers=AKSHTTPCustomFeatures=Microsoft.ContainerService/AKS-KedaPreview ' \
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:7955]: Does this feature still need the feature flag? If not, please remove this custom header

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 3]
```diff
@@ -7938,3 +7938,70 @@ def test_aks_nodepool_add_with_gpu_instance_profile(self, resource_group, resour
...
+        # create: without enable-keda
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --location={location} --ssh-key-value={ssh_key_value} --output=json'
+        self.cmd(create_cmd, checks=[
+            self.check('provisioningState', 'Succeeded'),
+            self.not_exists('workloadAutoScalerProfile.keda'),
+        ])
+
+        # update: enable-keda
+        update_cmd = 'aks update --resource-group={resource_group} --name={name} --yes --output=json ' \
+                     '--aks-custom-headers=AKSHTTPCustomFeatures=Microsoft.ContainerService/AKS-KedaPreview ' \
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:7988]: Also remove this header if the feature is not protected by AFEC anymore.

**Comment by FumingZhang** [comment]
PR blocked by SDK for containerservice generated according to latest stable API 2022-11-01 not released yet.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 2 above)
Comment (v-shenoy) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:7955]: Currently, it does. Once all other changes are done for GA, we will remove the feature flag. I'll send in another PR to change that here as well.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py (hunk 3 above)
Comment (v-shenoy) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:7988]: Please refer to my reply above.

**Comment by v-shenoy** [comment]
Closing as another PR (https://github.com/Azure/azure-cli/pull/25007) has been raised for the same.

**Pull request closed without merging**